src/middleware/authenticate.ts

```diff
@@ -0,0 +1,46 @@
+import jwt from 'jsonwebtoken';
+import { JWT_SECRET } from '../siteConstants';
+import { DeveloperDocument, Express } from '../types';
+import { createDbConnection, handleCloseConnection } from '../services/database';
+import { developerSchema } from '../models/Developer';
+
+interface PayloadType {
+  email: string;
+  iat?: number;
+  exp?: number;
+}
+
+export const authenticate: Express.MiddleWare = async (req, res, next) => {
+  let token: string | undefined;
+  // Check if token is set
+  if (req.headers.authorization && req.headers.authorization.startsWith('Bearer')) {
+    [, token] = req.headers.authorization.split(' ');
+  }
+
+  if (!token) {
+    return next(new Error('Unauthorized. Please login to continue.'));
+  }
+
+  let payload: PayloadType;
```

Comment:
nit: again, we should change this variable name since it's not really a "payload"